### PR TITLE
docs: add loopback types page

### PR DESCRIPTION
**Pull request by agnes512**
connects to https://github.com/strongloop/loopback-next/issues/5063

Screen Shot 2020-08-07 at 3 42 07 PM
Screen Shot 2020-08-07 at 3 42 27 PM
Screen Shot 2020-08-07 at 3 42 36 PM

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: docs/site/Model.md [hunk 1]
```diff
@@ -607,7 +607,7 @@ Here are general attributes for property definitions:
       <td>Yes</td>
       <td>String</td>
       <td>
-        Property type. Can be any type described in <a href="LoopBack-types.html">LoopBack types</a>.
+        Property type. Can be any type described in <a href="https://loopback.io/doc/en/lb4/LoopBack-types.html">LoopBack types</a>.
```
Comment (dhmlau) [inline docs/site/Model.md:610]: use `LoopBack-types.md`?

Comment (bajtos) [reply docs/site/Model.md:610]: Yes please, we should use local (relative) paths, not absolute URLs. The trick is to use `.md` extension, that way Jekyll converts `LoopBack-types.md` to `/doc/en/lb4/LoopBack-types.html` link.

Comment (agnes512) [reply docs/site/Model.md:610]: I don't think HTML takes local paths. From the preview, it has `http://127.0.0.1:4001/doc/en/lb4/LoopBack-types.md#object-types`. cc @bajtos

File: docs/site/LoopBack-types.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+lang: en
+title: 'LoopBack Types'
+keywords: LoopBack 4.0, LoopBack 4, Node.js, TypeScript, Type
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/LoopBack-types.html
+---
+
+## Overview
+
+When defining a model in LoopBack 4, property types is the important part. No
```
Comment (nabdelgadir) [inline docs/site/LoopBack-types.md:11]:
```suggestion
When defining a model in LoopBack 4, property types are the important part. No
```

File: docs/site/LoopBack-types.md [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+matter for defining properties with decorator `@property` or defining a model at
```
Comment (nabdelgadir) [inline docs/site/LoopBack-types.md:12]:
```suggestion
matter whether it is for defining properties with decorator `@property` or defining a model at
```

File: docs/site/LoopBack-types.md [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+runtime, you may want to specify the types in the definition. This is how we
+typically do:
```
Comment (nabdelgadir) [inline docs/site/LoopBack-types.md:14]:
```suggestion
typically specify types:
```

File: docs/site/LoopBack-types.md [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+```
+
+## Object types
+
+Use the Object type when you need to be able to accept values of different
+types, for example a string or an array.
+
+A model often has properties that consist of other properties. For example, the
+user model can have an `address` property that is in type `Address`, which has
+properties `street`, `city`, `state`, and `zipCode`. //LoopBack allows inline
```
Comment (nabdelgadir) [inline docs/site/LoopBack-types.md:171]: What is the `//` for?

File: docs/site/LoopBack-types.md [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+  // other props
+
+  @property({
+    type: 'object',
+  })
+  address: Address;
+}
+```
+
+The value of the address is the definition of the `address` type.
```
Comment (nabdelgadir) [inline docs/site/LoopBack-types.md:196]: Would it be `Address` instead of `address`?

File: docs/site/LoopBack-types.md [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+
+{% include important.html content="
+The user model has to reference the Address constructor or the model name - `'Address'`.
```
Comment (nabdelgadir) [inline docs/site/LoopBack-types.md:199]: Should we also mention you have to import it?

File: docs/site/LoopBack-types.md (hunk 5 above)
Comment (agnes512) [reply docs/site/LoopBack-types.md:171]: deleted

**Comment by agnes512** [comment]
> I have one question not sure if it worths mentioning in the doc. If I have something like:
```
  @property({
    type: 'number',
  })
  id: string;
```
> which one will take precedence?

@dhmlau I think the app takes `id: string` and `@property {type:..}` will be used by databases.

**Comment by agnes512** [comment]
> Please add the new page to lb4_sidebar, I think it should be nested under "Reference guides".

👍  Added it under "Reference guides".
Screen Shot 2020-08-11 at 11 28 06 AM

File: docs/site/LoopBack-types.md [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+title: 'LoopBack Types'
+keywords: LoopBack 4.0, LoopBack 4, Node.js, TypeScript, Type
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/LoopBack-types.html
+---
+
+## Overview
+
+When defining a model in LoopBack 4, property types are the important part. No
+matter where it is for defining properties with decorator `@property` or
```
Comment (nabdelgadir) [inline docs/site/LoopBack-types.md:12]:
```suggestion
matter whether it is for defining properties with decorator `@property` or
```

File: docs/site/LoopBack-types.md [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+    type: 'object',
+  })
+  address: Address;
+}
+```
+
+The value of the address is the definition of the `Address` type.
+
+{% include important.html content="
+The user model has to reference the Address constructor or import the model - `'Address'`.
```
Comment (nabdelgadir) [inline docs/site/LoopBack-types.md:198]:
```suggestion
The user model has to reference the Address constructor and import the model - `'Address'`.
```

File: docs/site/Model.md (hunk 1 above)
Comment (dhmlau) [reply docs/site/Model.md:610]: that's interesting. Looking at the [Server.md](https://raw.githubusercontent.com/strongloop/loopback-next/master/docs/site/Server.md), it is also referencing other md files.

Comment (dhmlau) [reply docs/site/Model.md:610]: Talked to @agnes512, it seems like things won't get converted as expected because it's inside a html table.

**Pull request merged**